pyproject.toml

```diff
@@ -5,16 +5,11 @@ requires = [
     "wheel",
     "Cython>=0.29.33",
 
-    # use oldest-supported-numpy which provides the oldest numpy version with
-    # wheels on PyPI
-    #
-    # see: https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg
-    "oldest-supported-numpy; python_version!='3.10' or platform_system!='Windows' or platform_python_implementation=='PyPy'",
-    # For CPython 3.10 under Windows, SciPy requires NumPy 1.22.3 while the
-    # oldest supported NumPy is defined as 1.21.6. We therefore need to force
-    # it for this specific configuration. For details, see
-    # https://github.com/scipy/scipy/blob/c58b608c83d30800aceee6a4dab5c3464cb1de7d/pyproject.toml#L38-L41
-    "numpy==1.22.3; python_version=='3.10' and platform_system=='Windows' and platform_python_implementation != 'PyPy'",
+    # Starting with NumPy 1.25, NumPy is (by default) as far back compatible
+    # as oldest-support-numpy was (customizable with a NPY_TARGET_VERSION
+    # define).  For older Python versions continue using oldest-support-numpy.
+    "numpy>=1.25; python_version>='3.9'",
```

Comment:
Will this wheel build work for Python 3.9 + NumPy 1.19.3?

This would be consistent with https://github.com/scipy/oldest-supported-numpy/blob/a94726bf419b4f7db1e077db97934f27d6d24ab0/setup.cfg#L58-L63

Reply:
Yes

Reply:
If that is the case, why is `numpy>=1.25` here. Is it for PyPy?

Reply:
I only added the "backcompat mode" in NumPy 1.25?!  Previously, we made you pin an ancient version for the silly reason that you *might* be using newer C-API, now you can just ensure you use a new enough version instead, which in principle allows us to push fixes for you also.  (and by defaulting to it removes the whole super tricky thing of needing to remember using oldest-support-numpy to begin with)

It isn't like it would be any use to write: `numpy>=1.19.3,!=1.20,!=1.21,!=1.22,!=1.23,!=1.4` which would be the true NumPy versions that will give you wheels that are generally compatible all the way back to 1.19.3 (on py3.9).